src/components/Modals/UploadDocumentModal.jsx

```diff
@@ -222,6 +223,7 @@ const UploadDocumentModal = ({ showModal, setShowModal }) => {
             </DialogActions>
           </FormControl>
         </form>
+        {processing && <LoadingAnimation loadingItem={file.name} />}
```

Comment:
Rather than it being condition within the FormSection component, I think it should be wrapping it as part of a ternary:

```
{ processing ? <LoadingAnimation /> : <FormSection>modal content here</FormSection> }
```

https://github.com/codeforpdx/PASS/assets/14917816/ce17ca2b-c7b3-4f9b-89bb-77294973f053

Reply:
So it would fully replace all of the content with the animation?

I prefer to stay with the Figma design. I think it looks a bit nicer and is less jarring than taking over the whole modal.

Reply:
The previous animation that's dangling at the bottom of the modal does not seem any better either. Might want to use a different kind of animation if not the full content

Reply:
Maybe we could look at a solution where the size of the modal does not change, which is jarring, and the loading animation displays over top of the modal with the content greyed out or removed. I don't mind the content being removed as long as the size of the modal doesn't change.

Reply:
> Maybe we could look at a solution where the size of the modal does not change, which is jarring, and the loading animation displays over top of the modal with the content greyed out or removed. I don't mind the content being removed as long as the size of the modal doesn't change.

We could try that. Or else the animation could temporarily replace one of the buttons (while retaining the size). Maybe "Upload File".

Thinking farther ahead, in #642 I mention possibly breaking down this whole process into several steps. That may or may not affect what we decide here.

Reply:
Hey @andycwilliams and @leekahung what do you think about something like this? 

Uploading Screen Recording 2024-06-28 at 3.59.39 PM.mov…

Reply:
The clip isn't showing. Think it might had not finished uploading

Reply:
Can't see your clip either but I just tested it locally. Not sure how it works for you, but this is what it's like for me:

<img width="348" alt="2024-06-28 (5)" src="https://github.com/codeforpdx/PASS/assets/83156697/a3cbb2c9-0a67-42c7-bee0-99ffa67ac8f7">

At least for me, the animation stretches out the modal/drawer to the point where it's not all visible. My preferred approach is to temporarily replace one of the elements with the animation rather than add it anywhere.

Seeing as we already have a line that states "File to upload" under "Take Photos" maybe we could just have the animation replace that? It might be worthwhile make that text a tad larger anyway. So "File to upload: [filename]" would become "Loading: [filename]". Or just replace the text with the animation, which would probably make it easier to keep the modal from changing size.

---

Another option is to use the `useNotification` hook instead. Maybe create a separate state for loading (right now there are only two: add and delete). I haven't thought about this approach very much.

---

As an aside, I tested clicking "Cancel" before loading was complete. It just closes the modal while uploading continues in the background. Do we want to make that button also cancel the uploading process? If not, it may be good to disable that button until the uploading process is done. This would be a separate issue of course.

Reply:
Uploading Screen Recording 2024-06-28 at 3.59.39 PM.mov…

Reply:
https://github.com/codeforpdx/PASS/assets/81979567/c4cda5ba-a8be-42a3-9e03-bd7c7669c738

Reply:
Ah, well that's pretty different! I'll take a closer look tomorrow.

Reply:
> @andycwilliams @leekahung  this is the clip I tried to post. I'm not sure why it wasn't working before. 
> 
> https://github.com/codeforpdx/PASS/assets/81979567/c4cda5ba-a8be-42a3-9e03-bd7c7669c738
> 
> 

I like this one 👍

Reply:
@andycwilliams Apologies - I did not push the changes I made to the code until just now.